### PR TITLE
Now reads from a YML file for configs.

### PR DESCRIPTION
**Pull request by btholt**
:eyeglasses: @ajacksified

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,14 +1,54 @@
...
+    config.vm.network "private_network", type: 'dhcp'
+  else
+    config.vm.network "private_network", ip: user_config[:private_ip]
   end
 
-  config.vm.box = "ubuntu/trusty64"
-  config.vm.box_url = "ubuntu/trusty64"
   config.vm.network :forwarded_port, guest: 4444, host: 4444
-  config.vm.provision :shell, :path => "sh/provision.sh"
+  config.vm.provision :shell, :path => "sh/provision.sh", :args => [
```
Comment (ajacksified) [inline Vagrantfile]: you'll also want https://github.com/reddit/reddit-vagrant/blob/master/Vagrantfile#L104-109 above this, to mount the src repos into vagrant.

cc @doriangray who knows more than me about this.

**Comment by btholt** [comment]
:nail_care: @ajacksified

**Comment by spladug** [comment]
(a wild pedant appears) not sure if the styleguide applies here, but these commit messages don't really fit with it: https://github.com/reddit/styleguide/tree/master/general-style-guidelines#git-workflow

**Comment by btholt** [comment]
Ah, it should apply. Thanks. I'll squash the commits and do a better one. Thanks @spladug

**Comment by btholt** [comment]
Closed in favor of #2 for better adherence to the reddit styleguide.

**Pull request closed without merging**